### PR TITLE
[opt] cct api refactor

### PR DESCRIPTION
**Pull request by lealzhan**
Re: #https://github.com/cocos/3d-tasks/issues/16826

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by yiwenxue — APPROVED** [review]

**Review by pandamicro — APPROVED** [review]

File: cocos/physics/framework/components/character-controllers/character-controller.ts [hunk 1]
```diff
@@ -295,23 +297,22 @@ export class CharacterController extends Eventify(Component) {
...
-     * Gets the position.
+     * Gets world position of center.
      * @zh
-     * 获取位置。
-     * @param out @zh 位置向量 @en The position vector
+     * 获取中心的世界坐标。
      */
-    public getPosition (out: Vec3) {
-        if (this._isInitialized) this._cct!.getPosition(out);
+    public get centerWorldPosition () {
```
Comment (SantyWang) [inline cocos/physics/framework/components/character-controllers/character-controller.ts]: Mark return value as `Readonly<Vec3>`

File: cocos/physics/framework/components/character-controllers/character-controller.ts [hunk 2]
```diff
@@ -320,9 +321,8 @@ export class CharacterController extends Eventify(Component) {
      * Gets the velocity.
      * @zh
      * 获取速度。
-     * @param out @zh 速度向量 @en The velocity vector
      */
-    public getVelocity (): Vec3 {
+    public get velocity (): Vec3 {
```
Comment (SantyWang) [inline cocos/physics/framework/components/character-controllers/character-controller.ts]: `Readonly<Vec3>`

File: cocos/physics/framework/physics-interface.ts [hunk 3]
```diff
@@ -263,15 +263,15 @@ export class CharacterControllerContact {
      * @zh
      * 碰撞中的角色控制器。
      */
-    selfCCT: any = null;
+    selfController: CharacterController = undefined!;
```
Comment (SantyWang) [inline cocos/physics/framework/physics-interface.ts]: Why use undefined

File: cocos/physics/framework/components/character-controllers/character-controller.ts (hunk 2 above)
Comment (lealzhan) [reply cocos/physics/framework/components/character-controllers/character-controller.ts]: ok

File: cocos/physics/framework/components/character-controllers/character-controller.ts (hunk 1 above)
Comment (lealzhan) [reply cocos/physics/framework/components/character-controllers/character-controller.ts]: ok

File: cocos/physics/framework/physics-interface.ts (hunk 3 above)
Comment (lealzhan) [reply cocos/physics/framework/physics-interface.ts]: Should we avoid using `any`?

Comment (shrinktofit) [reply cocos/physics/framework/physics-interface.ts]: Because he might want to reuse this class?

**Pull request merged**